tcp.rflx

```diff
@@ -0,0 +1,86 @@
+package TCP is
+
+    -- RFC 793
+
+    type Source_Port is mod 2**16;
+    type Destination_Port is mod 2**16;
+    type Sequence_Number is mod 2**32;
+    type Acknowledgment_Number is mod 2**32;
+    type Data_Offset is range 5 .. 2**4 - 1 with Size => 4;
+    type Reserved is mod 2**3;
+    type Window is mod 2**16;
+    type Checksum is mod 2**16;
+    type Urgent_Pointer is mod 2**16;
+
+    -- TIME_STAMP and WINDOW_SCALE options are described in RFC 7323
+    -- Selective Acknowledgment Options (SACK) are described in RFC 2018
+
+    type Kind is
+       (End_Of_Options_List    => 16#00#,
+        No_Operation           => 16#01#,
+        Maximum_Segment_Size   => 16#02#,
+        Window_Scale           => 16#03#,
+        SACK_Permitted         => 16#04#,
+        SACK                   => 16#05#,
+        Time_Stamp             => 16#08#)
+    with Size => 8;
+
+    type Length is range 2 .. 2**8 - 1 with Size => 8;
+
+    type Option is
+       message
+          Kind : Kind
+             then null
+                if Kind = End_Of_Options_List or Kind = No_Operation
+             then Length
+                if Kind = Maximum_Segment_Size
+                   or Kind = Window_Scale
+                   or Kind = SACK_Permitted
+                   or Kind = SACK
+                   or Kind = Time_Stamp;
+          Length : Length
+             then null
+                if Kind = SACK_Permitted
+             then Data
+                if Kind = Maximum_Segment_Size
+                   or Kind = Window_Scale
+                   or Kind = SACK
+                   or Kind = Time_Stamp;
+          Data : Opaque
+             with Size => (Length - 2) * 8;
```

Comment:
Since the new implementation of the TCP options the field Data becomes generally obsolete. 

This is changed in commit 4b515169953d78d669625c7fe9e0ebe219cc16e1